### PR TITLE
Feat/v feature discovery - initial concept

### PR DESCRIPTION
**Pull request by JamieCurnow**
## Description
A new 'Feature Discovery' component for Vuetify. 

## Motivation and Context
See #1171 

## How Has This Been Tested?
In playground

## Markup:

<details>

```vue
<template>
  <v-app>
    <v-content>
      <v-container >

        <v-layout row wrap justify-center>
          <v-flex xs6 text-xs-center>
            <v-card>
              <v-card-text>
                <v-layout row wrap justify-center>
                  <v-flex xs12 text-xs-center>
                    <h2>V Feature Discovery</h2>
                    <p>Currently overlay is not working since reqriting in TypeScript. Also should probably have a min height.</p>
                  </v-flex>
                </v-layout>

                <v-layout row wrap justify-center>
                  <v-spacer></v-spacer>
                  <v-switch label="top" v-model="top" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                  <v-switch label="left" v-model="left" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                  <v-switch label="flat" v-model="flat" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                </v-layout>
                <v-layout row wrap justify-center>
                  <v-spacer></v-spacer>
                  <v-switch label="hide-overlay" v-model="hideOverlay" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                  <v-switch label="persistent" v-model="persistent" style="flex-grow: 0"></v-switch>
                  <v-spacer></v-spacer>
                </v-layout>
                <v-layout row wrap justify-center>
                  <v-flex xs6 text-xs-center>
                    <v-text-field
                      type="number"
                      label="height"
                      v-model="height"
                    ></v-text-field>
                  </v-flex>
                </v-layout>
              </v-card-text>
              <v-card-actions>
                <v-spacer></v-spacer>
                <v-btn color="primary" @click="discover = !discover">Discover</v-btn>
                <v-spacer></v-spacer>
              </v-card-actions>
            </v-card>
          </v-flex>
        </v-layout>

        <v-feature-discovery
          class="white--text"
          color="primary"
          :top="top"
          :left="left"
          :hide-overlay="hideOverlay"
          :flat="flat"
          :persistent="persistent"
          :height="height"
          v-model="discover"
        >
          <h2 slot="title">Hey! New feature...</h2>
          <p slot="text">There's all sorts of new things that you can do with the great new features. Just click the buttons to check them out!</p>
          <div slot="actions">
            <v-btn color="white" depressed>Action</v-btn>
            <v-btn outline color="white" @click="discover = false">Lesser action</v-btn>
          </div>
        </v-feature-discovery>

      </v-container>
    </v-content>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
      discover: false,
      top: false,
      left: false,
      hideOverlay: false,
      flat: false,
      persistent: false,
      height: 400
    })
  }
</script>

```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

**Comment by robertBojor** [comment]
Is this going to come to production soon? Thanks! - Really looking forward for it.

File: src/stylus/components/_feature-discovery.styl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+v-feature-discovery($material)
+  color: $material.text.primary
+  background-color: $material.app-bar
+
+theme(v-feature-discovery, "v-feature-discovery")
+
+.v-feature-discovery
+  overflow: hidden;
+  transition: $primary-transition;
```
Comment (johnleider) [inline src/stylus/components/_feature-discovery.styl]: We don't actually lint the stylus files, but please remove the semi-colons

File: src/components/VFeatureDiscovery/VFeatureDiscovery.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    OuterContainerClasses (): object {
+      return {
+        [`justify-${this.left ? 'start' : 'end'}`]: true,
+        [`align-${this.top ? 'baseline' : 'end'}`]: true
+      }
+    },
+    innerContentClasses (): any {
+      return {
+        title: {
+          'mr-4': this.left && !this.top,
```
Comment (johnleider) [inline src/components/VFeatureDiscovery/VFeatureDiscovery.ts]: Avoid using helper classes

**Comment by johnleider** [comment]
This Pull Request is being closed due to unresolved merge conflicts. Once corrected we will re-open.

Thank you for your contribution and interest in improving Vuetify!

**Pull request closed without merging**